latch/types/metadata.py

```diff
@@ -621,39 +609,11 @@ class SnakemakeMetadata(LatchMetadata):
     """
 
     def validate(self):
-        def _contains_none_fields(t: Type, param: Any) -> bool:
-            if is_primitive_type(t) or t in {LatchFile, LatchDir}:
-                return param is None
-
-            if is_list_type(t):
-                if len(get_args(t)) == 0:
-                    raise ValueError(
-                        "Generic Lists are not supported - please specify a subtype,"
-                        " e.g. List[LatchFile]",
-                    )
-                list_typ = get_args(t)[0]
-                for val in param:
-                    if _contains_none_fields(list_typ, val):
-                        return True
-                return False
-
-            assert is_dataclass(t)
-            for field in fields(t):
-                if _contains_none_fields(field.type, getattr(param, field.name)):
-                    return True
-            return False
 
         for name, param in self.parameters.items():
             if param.default is None:
                 continue
-            if _contains_none_fields(param.type, param.default):
-                click.secho(
-                    " Non-None default parameters cannot contain None fields. Please"
-                    " remove default or set all fields to a non-None value for"
-                    f" parameter: {name}.  ",
-                    fg="red",
-                )
-                raise click.exceptions.Exit(1)
+            validate_snakemake_type(name, param.type, param.default)
```

Comment:
try-except this, `click.secho` the caught `ValueError`'s message, and then raise `click.exceptions.Exit(1)`